src/daft-core/src/array/ops/trigonometry.rs

```diff
@@ -56,3 +58,26 @@ where
         }
     }
 }
+
+macro_rules! impl_atan2_floating_array {
```

Comment:
You can likely rely on a generic instead of a macro for implementing the float arrays like here: https://github.com/Eventual-Inc/Daft/blob/395ebe8f406488b11eb65d71c0ff01ba76a61d76/src/daft-core/src/array/ops/trigonometry.rs#L40